### ISSUE TITLE
Receiving `COROUTINE_SUSPENDED` from `http.request`.

### ISSUE THREAD

**Issue by Kesanov** [korlibs/korio#131]
The following program prints `COROUTINE_SUSPENDED`

```kotlin
suspend fun foo(url: String, data: ByteArray? = null) = createHttpClient().request(Http.Method.GET, url).content
suspend fun main() = Korge(title="", width=800, height=600){
      println(foo(url))
}
```
It however works fine, if I remove the default argument `data: ByteArray? = null`.

**Comment by Josef-Vonasek** [comment]
When I close the windows, I also get an error:
```
Exception in thread "main" kotlinx.coroutines.CoroutinesInternalError: Fatal exception in coroutines machinery for AwaitContinuation(DispatchedContinuation[GameWindowCoroutineDispatcher, Continuation at com.soywiz.korio.async.AsyncThread.invoke(AsyncQueue.kt:75)@7a3793c7]){Cancelled}@42b3b079. Please read KDoc to 'handleFatalException' method and report this incident to maintainers
	at kotlinx.coroutines.DispatchedTask.handleFatalException(DispatchedTask.kt:144)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:115)
	at com.soywiz.korgw.GameWindowCoroutineDispatcher.executePending-_rozLdE(GameWindow.kt:128)
	at com.soywiz.korgw.GameWindowCoroutineDispatcher.close(GameWindow.kt:148)
	at com.soywiz.korgw.GameWindow.close(GameWindow.kt:331)
	at com.soywiz.korge.view.Views.close(Views.kt:161)
	at com.soywiz.korge.Korge$prepareViewsBase$8$1.invokeSuspend(Korge.kt:365)
	at com.soywiz.korge.Korge$prepareViewsBase$8$1.invoke(Korge.kt)
	at com.soywiz.korge.Korge$prepareViewsBase$8$1.invoke(Korge.kt)
	at com.soywiz.korio.async.AsyncExtKt$_launch$1.invokeSuspend(AsyncExt.kt:65)
	at com.soywiz.korio.async.AsyncExtKt$_launch$1.invoke(AsyncExt.kt)
	at com.soywiz.korio.async.AsyncExtKt$_launch$1.invoke(AsyncExt.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:55)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:112)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at com.soywiz.korio.async.AsyncExtKt._launch(AsyncExt.kt:63)
	at com.soywiz.korio.async.AsyncExtKt.launchImmediately(AsyncExt.kt:36)
	at com.soywiz.korge.Korge$prepareViewsBase$8.invoke(Korge.kt:364)
	at com.soywiz.korge.Korge$prepareViewsBase$8.invoke(Korge.kt:360)
	at com.soywiz.korev.EventDispatcher$Mixin.dispatch(EventDispatcher.kt:51)
	at com.soywiz.korgw.GameWindow.dispatchDestroyEvent(GameWindow.kt:746)
	at com.soywiz.korgw.awt.BaseAwtGameWindow.loop$suspendImpl(BaseAwtGameWindow.kt:600)
	at com.soywiz.korgw.awt.BaseAwtGameWindow.loop(BaseAwtGameWindow.kt)
	at com.soywiz.korge.Korge.invoke-V4YNAJM(Korge.kt:126)
	at com.soywiz.korge.Korge.invoke-V4YNAJM$default(Korge.kt:91)
	at MainKt.main(main.kt:23)
	at MainKt$main$3.invoke(main.kt)
	at MainKt$main$3.invoke(main.kt)
	at kotlin.coroutines.intrinsics.IntrinsicsKt__IntrinsicsJvmKt$createCoroutineUnintercepted$$inlined$createCoroutineFromSuspendFunction$IntrinsicsKt__IntrinsicsJvmKt$1.invokeSuspend(IntrinsicsJvm.kt:205)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:115)
	at kotlin.coroutines.jvm.internal.RunSuspendKt.runSuspend(RunSuspend.kt:19)
	at MainKt.main(main.kt)
Caused by: java.lang.ClassCastException: class kotlin.coroutines.jvm.internal.CompletedContinuation cannot be cast to class kotlinx.coroutines.internal.DispatchedContinuation (kotlin.coroutines.jvm.internal.CompletedContinuation and kotlinx.coroutines.internal.DispatchedContinuation are in unnamed module of loader 'app')
	at kotlinx.coroutines.CoroutineDispatcher.releaseInterceptedContinuation(CoroutineDispatcher.kt:108)
	at kotlin.coroutines.jvm.internal.ContinuationImpl.releaseIntercepted(ContinuationImpl.kt:118)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:39)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
Caused by: java.lang.ClassCastException: class kotlin.coroutines.jvm.internal.CompletedContinuation cannot be cast to class kotlinx.coroutines.internal.DispatchedContinuation (kotlin.coroutines.jvm.internal.CompletedContinuation and kotlinx.coroutines.internal.DispatchedContinuation are in unnamed module of loader 'app')

```

**Comment by Josef-Vonasek** [comment]
Also, decoding suddenly stopped working for some reason.
```
createHttpClient().request("https://raw.githubusercontent.com/korlibs/korio/master/README.md").content.readStringz()
```
Gives me
```
      ?R]o?}?? B?G???A??4M ?3'?l'??	??yp>??{N??y5?a?? ^?,s,??w???X??�?????tE?i?0???T??%7v????[?f?+HM??%$fuBH\S=k??+?-?+?f_ ??3?????X(??l%L+~??ZV?m?????`????|?y??0b{?tq?O????T??;???s?Uzg??mx?j????&+?IZ???q?1?V?t??{???�7\M???*?g{??????M??^\??}?g???N?????9???s?G???W?R??S??2?x???RC8?a???)?^?????????m0?@H&jQe?JM?????????????? 
```

**Comment by soywiz** [comment]
Can you try `Korio 2.2.1`, or `Korge 2.2.1.0`?

**Comment by Josef-Vonasek** [comment]
How do I bump the korge version, when using the korge gradle plugin?

**Comment by soywiz** [comment]
Edit the gradle.properties file and change the korgePluginVersion property to the right version

**Comment by Josef-Vonasek** [comment]
Latest version of korge plugin seems to be `2.1.1.6`.

**Comment by soywiz** [comment]
You can determine the latest version here: https://github.com/korlibs/korge-plugins/releases

**Comment by soywiz** [comment]
I'm closing this expecting this was fixed already in 2.4.7
